### PR TITLE
Dependency update and cleanup.

### LINKED ISSUE #988: Check that all sample library dependencies have been updated and fix breakages.
After merging #952 which updated library dependencies for most sample projects, CI has been knowingly broken. The libraries have taken a major version bump, breaking changes included, and that is what has broken samples.
I'm working through this right now.

### PR DESCRIPTION
**Pull request by amanda-tarafa**
Fixes #988

### CODE REVIEW

**Comment by amanda-tarafa** [comment]
Finally green, after having either fixed or skipped a lot of flakiness. Everything that was skipped has its corresponding issue.
@jskeet PTAL to the following commits which cover most of the types of code changes I had to make. Also I mostly only changed what was needed after the dependency update (lots of other changes are possible and even desirable but I can take care of that later). Note that I'm not using C# 8 because CI wasn't ready and I didn't want to delay this further.

- [Auth](https://github.com/GoogleCloudPlatform/dotnet-docs-samples/pull/1001/commits/0ad04274948f901bb258cf5a3386a7acc996a7f2)
- [Bigtable](https://github.com/GoogleCloudPlatform/dotnet-docs-samples/pull/1001/commits/ef125db4db9f64f6c0dae8054fd4acf23ad125c2)
 - [Firestore](https://github.com/GoogleCloudPlatform/dotnet-docs-samples/pull/1001/commits/de7d6839a867b8e52a54e67b4779317794f00722)
 - [Talent V4](https://github.com/GoogleCloudPlatform/dotnet-docs-samples/pull/1001/commits/25e27ff0859aa8a9889da2b2b9c35c3dd05fcd16)
- [Logging](https://github.com/GoogleCloudPlatform/dotnet-docs-samples/pull/1001/commits/f438f67ab90365d39303000f56f8b6a42f75e378)

**Comment by jskeet** [comment]
Congrats! Will look at this first thing on Monday.

File: applications/leaderboard/Leaderboard/Leaderboard.csproj [hunk 1]
```diff
@@ -1,14 +1,12 @@
-<Project Sdk="Microsoft.NET.Sdk">
+﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFramework>netcoreapp2.0</TargetFramework>
+    <TargetFramework>netcoreapp2.1</TargetFramework>
```
Comment (jskeet) [inline applications/leaderboard/Leaderboard/Leaderboard.csproj:5]: Let's consider updating everything to 3.1 *after* this PR is done. I think it's been out for long enough that we can assume people can use it, at least for samples.

File: applications/leaderboard/LeaderboardTest/LeaderboardTest.csproj [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
-    <PackageReference Include="Google.Cloud.Spanner.Data" Version="2.1.0" />
-    <PackageReference Include="Google.Cloud.Spanner.V1" Version="2.1.0" />
-    <PackageReference Include="JUnitTestLogger" Version="0.6.0" />
-    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.9.0" />
-    <PackageReference Include="xunit" Version="2.2.0" />
-    <PackageReference Include="xunit.runner.visualstudio" Version="2.2.0" />
+    <PackageReference Include="JUnitTestLogger" Version="1.1.0" />
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.5.0" />
+    <PackageReference Include="xunit.runner.visualstudio" Version="2.4.1">
+      <PrivateAssets>all</PrivateAssets>
```
Comment (jskeet) [inline applications/leaderboard/LeaderboardTest/LeaderboardTest.csproj:13]: Why are the asset specifications needed? If we can get away with just including the reference, I think that would be simpler.
Not to worry if we really need them. Although we could potentially use `PrivateAssets="all" IncludeAssets="all"` as attributes in the element.

We might also want to configure having a common "CommonDependencies.xml" or similar that gets included, so we only need to change one file in future.
These are suggestions for a later PR though :)

File: applications/sudokumb/WebLib/DatastoreCounter.cs [hunk 3]
```diff
@@ -184,8 +184,8 @@ public async Task CondenseOldCounters(CancellationToken cancellationToken)
                 Projection = { "__key__" }
             };
             var lazyResults = _datastore.RunQueryLazilyAsync(query,
-                callSettings: callSettings).GetEnumerator();
-            while (await lazyResults.MoveNext())
+                callSettings: callSettings).GetAsyncEnumerator(cancellationToken);
```
Comment (jskeet) [inline applications/sudokumb/WebLib/DatastoreCounter.cs:187]: If we move to .NET Core 3.1 we can use await foreach, which would be much nicer.

File: auth/AuthSample/Program.cs [hunk 4]
```diff
@@ -335,12 +335,12 @@ public class CloudLibrary : AuthLibrary
         // method.
         public object AuthExplicit(string projectId, string jsonPath)
         {
-            var credential = GoogleCredential.FromFile(jsonPath)
-                .CreateScoped(LanguageServiceClient.DefaultScopes);
-            var channel = new Grpc.Core.Channel(
-                LanguageServiceClient.DefaultEndpoint.ToString(),
-                credential.ToChannelCredentials());
-            var client = LanguageServiceClient.Create(channel);
+            LanguageServiceClientBuilder builder = new LanguageServiceClientBuilder
```
Comment (jskeet) [inline auth/AuthSample/Program.cs:338]: I'd probably do all of this in a single line:

```csharp
LanguageServiceClient client = new LanguageServiceClientBuilder { CredentialsPath = jsonPath }.Build();
```

That's more the kind of code I expect our users to write.

File: auth/AuthSample/Program.cs [hunk 5]
```diff
@@ -351,11 +351,14 @@ public object AuthExplicit(string projectId, string jsonPath)
         // method.
         public object AuthExplicitComputeEngine(string projectId)
         {
-            var credential = GoogleCredential.FromComputeCredential();
-            var channel = new Grpc.Core.Channel(
-                LanguageServiceClient.DefaultEndpoint.ToString(),
-                credential.ToChannelCredentials());
-            var client = LanguageServiceClient.Create(channel);
+            LanguageServiceClientBuilder builder = new LanguageServiceClientBuilder
```
Comment (jskeet) [inline auth/AuthSample/Program.cs:354]: In this case, I'd keep it multi-line, but still just have a `client` variable, calling `Build()` at the end.

File: firestore/api/GetData/Program.cs [hunk 6]
```diff
@@ -204,9 +204,11 @@ private static async Task GetCollections(string project)
             FirestoreDb db = FirestoreDb.Create(project);
             // [START fs_get_collections]
             DocumentReference cityRef = db.Collection("cities").Document("SF");
-            IList<CollectionReference> subcollections = await cityRef.ListCollectionsAsync().ToList();
-            foreach (CollectionReference subcollectionRef in subcollections)
+            IAsyncEnumerable<CollectionReference> subcollections = cityRef.ListCollectionsAsync();
+            IAsyncEnumerator<CollectionReference> subcollectionsEnumerator = subcollections.GetAsyncEnumerator(default);
+            while (await subcollectionsEnumerator.MoveNextAsync())
```
Comment (jskeet) [inline firestore/api/GetData/Program.cs:209]: More await foreach stuff. If we can use C# 8, we should do a search for MoveNextAsync.

File: irm/api/Irm.sln [hunk 7]
```diff
@@ -0,0 +1,25 @@
+﻿
+Microsoft Visual Studio Solution File, Format Version 12.00
```
Comment (jskeet) [inline irm/api/Irm.sln:2]: Do we want an IRM sample at all?

File: logging/api/QuickStart/QuickStart.cs [hunk 8]
```diff
@@ -56,8 +55,10 @@ public static void Main(string[] args)
             // This sample uses resource type 'global' causing log entries to appear in the 
             // "Global" resource list of the Developers Console Logs Viewer:
             //  https://console.cloud.google.com/logs/viewer
-            MonitoredResource resource = new MonitoredResource();
-            resource.Type = "global";
+            MonitoredResource resource = new MonitoredResource
```
Comment (jskeet) [inline logging/api/QuickStart/QuickStart.cs:58]: Maybe collapse to one line?

File: monitoring/api/QuickStart/QuickStart.cs [hunk 9]
```diff
@@ -37,33 +38,47 @@ public static void Main(string[] args)
             // Initialize request argument(s).
             ProjectName name = new ProjectName(projectId);
 
-            // Prepare a data point. 
-            Point dataPoint = new Point();
-            TypedValue salesTotal = new TypedValue();
-            salesTotal.DoubleValue = 123.45;
-            dataPoint.Value = salesTotal;
+            // Prepare a data point.
```
Comment (jskeet) [inline monitoring/api/QuickStart/QuickStart.cs:41]: Collapse each of these to a single line? That'll make it *much* briefer.

File: pubsub/api/PubsubSample/Program.cs [hunk 10]
```diff
@@ -501,17 +498,11 @@ public static object ListProjectTopics(PublisherServiceApiClient publisher, stri
         public static PublisherServiceApiClient CreatePublisherWithServiceCredentials(
             string jsonPath)
         {
-            GoogleCredential googleCredential = null;
-            using (var jsonStream = new FileStream(jsonPath, FileMode.Open,
-                FileAccess.Read, FileShare.Read))
+            PublisherServiceApiClientBuilder builder = new PublisherServiceApiClientBuilder
```
Comment (jskeet) [inline pubsub/api/PubsubSample/Program.cs:501]: See earlier comments about builder variables. (Only suggestions though.)

File: speech/api/Recognize/Recognize.cs [hunk 11]
```diff
@@ -404,15 +401,11 @@ static object RecognizeWithContext(string filePath, IEnumerable<string> phrases)
 
         static object SyncRecognizeWithCredentials(string filePath, string credentialsFilePath)
         {
-            GoogleCredential googleCredential;
-            using (Stream m = new FileStream(credentialsFilePath, FileMode.Open))
+            SpeechClientBuilder builder = new SpeechClientBuilder
```
Comment (jskeet) [inline speech/api/Recognize/Recognize.cs:404]: Builder comments.

File: dialogflow/api/DialogflowSamples/ContextManagement.cs [hunk 12]
```diff
@@ -121,7 +127,11 @@ public class DeleteAllOptions : OptionsWithProjectIdAndSessionId { }
         public static int DeleteAllContexts(string projectId, string sessionId)
         {
             var client = ContextsClient.Create();
-            client.DeleteAllContexts(new SessionName(projectId, sessionId));
+            var deleteAllContextsRequests = new DeleteAllContextsRequest
+            {
+                Parent = SessionName.FromProjectSession(projectId, sessionId).ToString()
+            };
+            client.DeleteAllContexts(deleteAllContextsRequests);
```
Comment (jskeet) [inline dialogflow/api/DialogflowSamples/ContextManagement.cs:134]: As of Friday, you can pass the session name in directly. (Will need to do another release, of course...)

**Comment by amanda-tarafa** [comment]
Thanks!
And yes, agree with your comments.

- I did tried to use netcoreapp3.1 and C# 8 (await foreach included), but CI needs tweaking for that and I didn't want to delay this further. (just created #1013).

- IRM sample were deleted in #1005, and I think I rebased after. Will rebase now then just in case.

- For fixing Dialogflow resource name usage there's #1003 that I will unblock after there's a release. No rush on my part, I have a few other things I need to get to first anyway.

**Pull request merged**